### PR TITLE
Edit subdomain char limit

### PR DESCRIPTION
**Pull request by zaelgohary**
### Changes

- Validate subdomain
- Edit subdomain to include prefix by default
- Edit max char subdomain as the validation function used gets a name argument with max 30 char
- Edit max contract name in get_name_contract to 30
- Edit tooltip & table name

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2564

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by MohamedElmdary** [comment]
Shouldn't we keep deployment as before with oldPrefix ?

File: packages/playground/src/utils/validators.ts [hunk 1]
```diff
@@ -771,3 +772,11 @@ export async function isValidStellarAddress(target: string): Promise<RuleReturn>
     return { message };
   }
 }
+
+export async function isValidSubdomain(grid: GridClient, subdomain: string) {
+  const valid = await grid.contracts.get_name_contract({ name: subdomain });
+  console.log({ valid });
```
Comment (AhmedHanafy725) [inline packages/playground/src/utils/validators.ts:778]: please remove the debug logs

File: packages/playground/src/utils/validators.ts [hunk 2]
```diff
@@ -771,3 +772,11 @@ export async function isValidStellarAddress(target: string): Promise<RuleReturn>
...
+  if (subdomain && !!valid) {
+    return { message: "Subdomain  already exists." };
```
Comment (AhmedHanafy725) [inline packages/playground/src/utils/validators.ts:780]:
```suggestion
    return { message: "Subdomain is already taken." };
```

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: packages/playground/src/components/manage_gateway_dialog.vue [hunk 3]
```diff
@@ -128,15 +118,12 @@
                   validators.isAlphanumeric('Subdomain should consist of letters and numbers only.'),
                   subdomain => validators.isAlpha('Subdomain must start with alphabet char.')(subdomain[0]),
                   validators.minLength('Subdomain must be at least 4 characters.', 4),
-                  subdomain =>
-                    validators.maxLength(
-                      `Subdomain cannot exceed ${35 - prefix.length} characters.`,
-                      35 - prefix.length,
-                    )(subdomain),
+                  subdomain => validators.maxLength('Subdomain cannot exceed 15 characters.', 15)(subdomain),
```
[the remark this answers is not in the document]
Comment (zaelgohary) [reply packages/playground/src/components/manage_gateway_dialog.vue:121]: I had to change the limit as the validation for name in get_name_contract is 15 chars max. so, I either loosen the validation from the Dashboard or the Grid Client.

[the remark this answers is not in the document]
Comment (Mahmoud-Emad) [reply packages/playground/src/components/manage_gateway_dialog.vue:121]: Hello, What is the status here? :)

[the remark this answers is not in the document]
Comment (AhmedHanafy725) [reply packages/playground/src/components/manage_gateway_dialog.vue:121]: please change it in the grid client

File: packages/grid_client/src/modules/models.ts [hunk 4]
```diff
@@ -340,7 +340,7 @@ class GetServiceContractModel {
   @Expose() @IsInt() @Min(1) serviceId: number;
 }
 class NameContractGetModel {
-  @Expose() @IsString() @IsNotEmpty() @IsAlphanumeric() @MaxLength(NameLength) name: string;
+  @Expose() @IsString() @IsNotEmpty() @IsAlphanumeric() @MaxLength(30) name: string;
```
Comment (AhmedHanafy725) [inline packages/grid_client/src/modules/models.ts:343]: to be like the gateways model make it `NameLength + 20`

**Review by MohamedElmdary — APPROVED** [review]

**Review by samaradel — APPROVED** [review]

**Pull request merged**